### PR TITLE
[ASDisplayNode] Remove node from supernode before adding it as a subview

### PR DESCRIPTION
**Pull request by rcancro**
If a node is being added as a subview to a UIVIew, we must make sure to remove it as a subnode of its supernode.

### CODE REVIEW

File: AsyncDisplayKit/ASDisplayNode.mm [hunk 1]
```diff
@@ -3857,6 +3857,9 @@ - (void)addSubnode:(ASDisplayNode *)subnode
     if (selfNode) {
       [selfNode addSubnode:subnode];
     } else {
+      if (subnode.supernode) {
```
Comment (appleguy) [inline AsyncDisplayKit/ASDisplayNode.mm]: @rcancro thanks for submitting this!  I'm curious how you found it.

Could you also fix the CALayer -addSubnode: category method that is just beneath this?

**Comment by garrettmoon** [comment]
test this please

**Comment by rcancro** [comment]
I found it when working on a custom transition. I noticed that if I added an ASDisplayNode to the transition context's `containerView` that the ASDisplaynode did not appear unless I explicitly called `removeFromSupernode`

**Comment by appleguy** [comment]
@rcancro phenomenal fix, thank you!!

**Pull request merged**